### ISSUE TITLE
Button - new type 'danger'

### ISSUE THREAD

**Issue by hohnzy** [infor-design/enterprise-wc#985]
**Is your feature request related to a problem or use case? Please describe.**
We'd like to have a 'danger' type button. We want to use it before deleting crucial customers data for example from S3 buckets etc.

**Describe the solution you'd like**
Probably red backgrounded button (ruby-60/ruby-80 on hover?).
![image](https://user-images.githubusercontent.com/11279863/200034079-7606b0d4-330a-4b2d-8e8e-908682e26b9c.png)

**Describe alternatives you've considered**
We can use css restyling, something like this works just fine
```
ids-button.danger-button::part(button) {
  background-color: var(--ids-color-palette-ruby-60);
  border-color: var(--ids-color-palette-ruby-60);

  &:hover {
    background-color: var(--ids-color-palette-ruby-80);
  }

  &:focus {
    box-shadow: 0 0 0 2px var(--ids-color-palette-white), 0 0 0 3px var(--ids-color-palette-ruby-60), 0 0 4px 3px var(--ids-color-palette-ruby-60);
  }
}
```

**Additional context**
Data Fabric application

**Comment by tmcconechy** [comment]
Main Color: DA1217
Disabled Color: ?
Hover Color: ?

**Comment by tmcconechy** [comment]
Design: https://www.figma.com/file/VcRDGPFqlp3MJ46mEWkGPf/Buttons?node-id=701%3A1222&t=2eRdbDDXoBLYCBwt-1

**Comment by hohnzy** [comment]
Looks good :)
Do we want to have the blue focus?

**Comment by tmcconechy** [comment]
@hohnzy you mean the focus state might be red instead of blue?

**Comment by hohnzy** [comment]
That's just my opinion, but the red/blue contrast is a bit weird to my eyes.

**Comment by tmcconechy** [comment]
yeah i agree - just making sure we are thinking the same thing. I'll talk to design.

**Comment by tmcconechy** [comment]
Almost a duplicate of #1046
